File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int l = 0; 
        int r = nums.length - 1;
        int i = nums.length - 1;
        int[] res = new int[nums.length];

        while(l<=r){
            if(nums[l]*nums[l] > nums[r]*nums[r]){
                res[i] = nums[l]*nums[l];
                l+=1;
            }
            else{
                res[i] = nums[r]*nums[r];
                r-=1;
            }
        i-=1;

        }

    return res;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l =0 
        r = len(nums) - 1
        res = [0] * len(nums)
        i = len(nums) - 1
        while l<=r:
            if nums[l]*nums[l] < nums[r]*nums[r]:
                # res.append(nums[r]**2)
                res[i] = nums[r]*nums[r]
                r-=1
            else:
                # res.append(nums[l]**2)
                res[i] = nums[l]*nums[l]
                l+=1
            i-=1
        return res 
        
        